File: dropwizard_2.9.1/src/test/scala/com/yammer/dropwizard/examples/SayCommand.scala
package com.yammer.dropwizard.examples

import com.yammer.dropwizard.cli.ConfiguredCommand
import org.apache.commons.cli.{Options, CommandLine}
import com.codahale.logula.Logging
import com.yammer.dropwizard.AbstractService

class SayCommand extends ConfiguredCommand[ExampleConfiguration]("say", "Prints out the saying to console") with Logging {
  override def getOptions = {
    val options = new Options
    options.addOption("v", "verbose", false, "yell it a lot")
    options
  }

  protected def run(service: AbstractService[ExampleConfiguration],
                    configuration: ExampleConfiguration,
                    params: CommandLine) {
    for (i <- 1 to (if (params.hasOption("verbose")) 10 else 1)) {
      log.warn(configuration.saying)
    }
  }
}

File: dropwizard/src/test/java/com/yammer/dropwizard/example/SayCommand.java
package com.yammer.dropwizard.example;

import com.yammer.dropwizard.AbstractService;
import com.yammer.dropwizard.cli.ConfiguredCommand;
import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.Options;

@SuppressWarnings("NullableProblems")
public class SayCommand extends ConfiguredCommand<ExampleConfiguration> {
    public SayCommand() {
        super(ExampleConfiguration.class, "say", "Prints out the saying to console");
    }

    @Override
    public Options getOptions() {
        return new Options().addOption("v", "verbose", false, "yell it a lot");
    }

    @Override
    protected void run(AbstractService<ExampleConfiguration> service,
                       ExampleConfiguration configuration,
                       CommandLine params) throws Exception {
        final String saying = configuration.getSaying();
        int max = 1;
        if (params.hasOption("verbose")) {
            max = 10;
        }

        for (int i = 0; i < max; i++) {
            System.err.println(saying);
        }
    }
}
